Adaptors/S3/src/ObjectStorage.cs

```diff
@@ -169,17 +168,37 @@ await taskList.WhenAll()
                                       {
                                         Key = $"{objectStorageName_}{key}_count",
                                       },
-                                    })
-                            .ToList();
-    var multiObjectDeleteRequest = new DeleteObjectsRequest
-                                   {
-                                     BucketName = bucketName_,
-                                     Objects    = keyList,
-                                   };
+                                    });
 
-    var deleteObjectsResponse = await s3Client_.DeleteObjectsAsync(multiObjectDeleteRequest)
-                                               .ConfigureAwait(false);
-    return deleteObjectsResponse.DeletedObjects.Count == valuesCount + 1;
+    var deletedItem = 0;
+    try
+    {
+      var multiObjectDeleteRequest = new DeleteObjectsRequest
```

Comment:
You need to create this object in the select.
You tried to use thread safe access for the add but not for the object list.

Reply:
indeed

Reply:
updated